crates/ty/CONTRIBUTING.md

```diff
@@ -0,0 +1,127 @@
+# Contributing to ty
+
+Welcome! We're happy to have you here. Thank you in advance for your contribution to ty.
+
+> [!NOTE]
+>
+> This guide is for ty. If you're looking to contribute to Ruff, please see
+> [the Ruff contributing guide](../../CONTRIBUTING.md).
+
+## The Basics
+
+Ty welcomes contributions in the form of pull requests.
+
+For small changes (e.g., bug fixes), feel free to submit a PR.
+
+For larger changes (e.g. new diagnostics, new functionality, new configuration options), consider
+creating an [**issue**](https://github.com/astral-sh/ty/issues) outlining your proposed change.
+You can also join us on [Discord](https://discord.com/invite/astral-sh) to discuss your idea with the
+community. We've labeled [beginner-friendly tasks](https://github.com/astral-sh/ty/issues?q=is%3Aissue+is%3Aopen+label%3A%22help+wanted%22)
+in the issue tracker, along with [bugs](https://github.com/astral-sh/ty/issues?q=is%3Aissue+is%3Aopen+label%3Abug)
+that are ready for contributions.
+
+### Prerequisites
+
+Ty is written in Rust. You'll need to install the
+[Rust toolchain](https://www.rust-lang.org/tools/install) for development.
+
+You'll need [uv](https://docs.astral.sh/uv/getting-started/installation/) (or `pipx` and `pip`) to
+run Python utility commands.
+
+You can optionally install pre-commit hooks to automatically run the validation checks
+when making a commit:
+
+```shell
+uv tool install pre-commit
+pre-commit install
+```
+
+We recommend [nextest](https://nexte.st/) to run Ruff's test suite (via `cargo nextest run`),
+though it's not strictly necessary:
+
+```shell
+cargo install cargo-nextest --locked
+```
+
+Throughout this guide, any usages of `cargo test` can be replaced with `cargo nextest run`,
+if you choose to install `nextest`.
+
+### Development
+
+After cloning the repository, run ty locally from the repository root with:
+
+```shell
+cargo run --bin ty -- check --project /path/to/project/
+```
+
+Prior to opening a pull request, ensure that your code has been auto-formatted,
+and that it passes both the lint and test validation checks:
+
+```shell
+cargo clippy --workspace --all-targets --all-features -- -D warnings  # Rust linting
+cargo test  # Rust testing
+uvx pre-commit run --all-files --show-diff-on-failure  # Rust and Python formatting, Markdown and Python linting, etc.
+```
+
+These checks will run on GitHub Actions when you open your pull request, but running them locally
+will save you time and expedite the merge process.
+
+If you're using VS Code, you can also install the recommended [rust-analyzer](https://marketplace.visualstudio.com/items?itemName=rust-lang.rust-analyzer) extension to get these checks while editing.
+
+Include the text `[ty]` at the beginning of your pull request title, to distinguish ty pull requests
+from Ruff ones.
+
+Your pull request will be reviewed by a maintainer, which may involve a few rounds of iteration
+prior to merging.
+
```

Comment:
I think we should add a link to https://github.com/astral-sh/ruff/blob/bdccb37b4ae17606015d4c03e04f1dd6f1d6f4da/crates/ty/docs/tracing.md either at the end of this section (like "Refer to the tracing documentation for instructions on how to debug ty") or add a new "Debugging ty" section that provides a link to the tracing docs.